domenetjenester/uttak/src/main/java/no/nav/foreldrepenger/domene/uttak/uttaksgrunnlag/fp/MorsJustering.java

```diff
@@ -38,20 +41,27 @@ class MorsJustering implements ForelderFødselJustering {
     public List<OppgittPeriodeEntitet> justerVedFødselEtterTermin(List<OppgittPeriodeEntitet> oppgittePerioder) {
         oppgittePerioder = fyllHull(oppgittePerioder);
         var ikkeFlyttbarePerioder = ikkeFlyttbarePerioder(oppgittePerioder);
-        var hullPåSluttenAvTidsperiodeForbeholdtMor = ikkeFlyttbarePerioder.stream()
-            .filter(JusterPeriodeHull.class::isInstance)
-            .filter(p -> tilOgMed(gammelFamiliehendelse.plusDays(1)).isEqual(p.getFom()))
-            .findFirst();
         var virkedagerSomSkalSkyves = beregnAntallLedigeVirkedager(gammelFamiliehendelse, nyFamiliehendelse, ikkeFlyttbarePerioder);
-
         List<OppgittPeriodeEntitet> justertePerioder = new ArrayList<>();
         for (var oppgittPeriode : oppgittePerioder) {
-            var virkedagerSomSkalSkyvePeriode = hullPåSluttenAvTidsperiodeForbeholdtMor
-                .filter(hull -> !oppgittPeriode.getTom().isBefore(hull.getFom()))
-                .map(hull -> virkedagerSomSkalSkyves - Virkedager.beregnAntallVirkedager(hull.getFom(), hull.getTom()))
-                .orElse(virkedagerSomSkalSkyves);
-            var justert = flyttPeriodeTilHøyre(oppgittPeriode, virkedagerSomSkalSkyvePeriode, ikkeFlyttbarePerioder);
-            justertePerioder.addAll(justert);
+            if (virkedagerSomSkalSkyves > 0) {
+                var justert = flyttPeriodeTilHøyre(oppgittPeriode, virkedagerSomSkalSkyves, ikkeFlyttbarePerioder);
+                if (oppgittPeriode.getPeriodeType().equals(UttakPeriodeType.FELLESPERIODE) && erPeriodeFlyttbar(oppgittPeriode)) {
+                    var justertFellesperiode = reduserJusterteFellesperioder(justert, virkedagerSomSkalSkyves);
+                    virkedagerSomSkalSkyves -= justert.stream().mapToInt(p -> Virkedager.beregnAntallVirkedager(p.getFom(), p.getTom())).sum() - justertFellesperiode.stream().mapToInt(p -> Virkedager.beregnAntallVirkedager(p.getFom(), p.getTom())).sum();
```

Comment:
Få utledingen her inn i egen metode med forklarende navn hadde vært nice. AntallDagerFellesperiode ellerno

Reply:
Enig. Har gjort det nå for begge plassene hvor virkedagerSomSkalSkyves trekkes fra.